### ISSUE TITLE
`--grad-accum`, `--deepspeed` and `--logging-backend` are accepted by the GRPO CLI and have no effect

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#79]
Three flags on `thinkrl grpo` are parsed and then never applied.

**`--grad-accum`** appears exactly twice in `cli/grpo.py`: the option declaration at line 51, and line 217, where it is written into the `wandb.init(config=...)` dictionary. It is not a parameter of `GRPOTrainer.__init__` or `GRPOTrainer.train`:

```
GRPOTrainer.train signature: (self, steps: int = 1000, batch_size: int = 4, log_interval: int = 10)
call site: trainer.train(steps=total_steps, batch_size=per_device_train_batch_size)
```

It cannot reach the optimizer. The value does survive into the Weights and Biases run config, so a run is recorded as though gradient accumulation was applied when it was not.

**`--deepspeed`** is declared at line 58 and echoed at line 103, and is read nowhere else. `configs/ds_zero2.json` ships in the repository and `README.md:214` lists "Distributed Training: DeepSpeed integration (ZeRO-2/3)" as production-ready, so passing a ZeRO config is a natural thing to try. It produces a plain single-process run.

**`--logging-backend`** defaults to `"tensorboard"` at lines 69-71, and the only consumer is line 206, `if logging_backend == "wandb"`. The default value therefore logs nothing, `tensorboard` is a no-op, and an unrecognised value is accepted without complaint. A TensorBoard logger already exists in the package and is covered by `tests/test_logging_module/test_tensorboard.py`; it is simply not wired to this flag.

**Suggested fix**

For `--logging-backend`, wire the existing TensorBoard logger and reject unknown values. For `--grad-accum` and `--deepspeed`, either implement them in `GRPOTrainer` or remove the flags until they are supported. `thinkrl/training/sft_trainer.py` already implements gradient accumulation and a DeepSpeed engine path, so there is an in-repository pattern to follow for the first option.

I am happy to open a PR for whichever direction you prefer.
